nomenclature/processor/iamc.py

```diff
@@ -24,12 +24,13 @@ def validate_with_definition(self, dsd: DataStructureDefinition) -> None:
 
         # check for filter-items that are not defined in the codelists
         for dimension in IAMC_IDX:
-            if codelist := getattr(dsd, dimension, None) is None:
+            codelist = getattr(dsd, dimension, None)
+            if codelist is None:
                 continue
             if invalid := codelist.validate_items(getattr(self, dimension) or []):
```

Comment:
Not from this PR but you can directly give `getattr` the default/fallback:
```suggestion
            if invalid := codelist.validate_items(getattr(self, dimension, [])):
```